app/lib/actions/katello/host/update.rb

```diff
@@ -30,6 +30,10 @@ def plan(host, consumer_params = nil)
           plan_self(:hostname => host.name)
         end
 
+        def resource_locks
+          :link
```

Comment:
Yes, you can add another lock type for the host here https://github.com/Katello/katello/blob/master/app/models/katello/system.rb#L298 (available_locks method) an then here using this name. This would mean, that the lock would be exclusive only for this action and not influence the rest.